File: app/src/test/resources/testcode-go/class_usage_patterns.go
package main

// BaseStruct is a test struct
type BaseStruct struct {
	Value int
}

// StaticMethod is a function associated with BaseStruct
func StaticMethod() string {
	return "static"
}

// InstanceMethod is a method on BaseStruct
func (b *BaseStruct) InstanceMethod() string {
	return "instance"
}

// ExtendedStruct embeds BaseStruct
type ExtendedStruct struct {
	BaseStruct
	field *BaseStruct
}

// NewExtendedStruct creates a new ExtendedStruct
func NewExtendedStruct(param *BaseStruct) *ExtendedStruct {
	// Struct initialization
	local := BaseStruct{Value: 42}

	// Pointer type
	ptr := &BaseStruct{Value: 10}

	return &ExtendedStruct{
		BaseStruct: local,
		field:      ptr,
	}
}

// ProcessBase takes and returns BaseStruct
func (e *ExtendedStruct) ProcessBase(input *BaseStruct) *BaseStruct {
	return input
}

// GetSlice returns a slice of BaseStruct
func (e *ExtendedStruct) GetSlice() []*BaseStruct {
	return []*BaseStruct{{Value: 1}}
}

func usageExample() {
	// Variable with type
	var myBase BaseStruct
	var ptr *BaseStruct = &BaseStruct{Value: 5}
}

File: brokk-shared/src/test/resources/testcode-rs/class_usage_patterns.rs
/// Test file to verify Rust class usage patterns

pub struct BaseStruct {
    value: i32,
}

impl BaseStruct {
    pub fn new(value: i32) -> Self {
        BaseStruct { value }
    }

    pub fn static_method() -> String {
        "static".to_string()
    }
}

pub trait BaseTrait {
    fn process(&self);
}

// Trait implementation
impl BaseTrait for BaseStruct {
    fn process(&self) {
        println!("Processing");
    }
}

pub struct ExtendedStruct {
    // Field with type annotation
    base: BaseStruct,
}

impl ExtendedStruct {
    pub fn new(param: BaseStruct) -> Self {
        // Struct initialization
        let local = BaseStruct { value: 42 };

        // Associated function call
        let _result = BaseStruct::static_method();

        ExtendedStruct { base: param }
    }

    // Method with type annotations
    pub fn process_base(&self, input: BaseStruct) -> BaseStruct {
        input
    }

    // Generic usage
    pub fn get_vec(&self) -> Vec<BaseStruct> {
        vec![BaseStruct::new(1)]
    }
}

// Variable with type annotation
pub fn use_base() {
    let my_base: BaseStruct = BaseStruct::new(10);
}
